packages/odyssey-react-mui/src/Autocomplete.tsx

```diff
@@ -292,6 +319,83 @@ const Autocomplete = <
       testId,
     ]
   );
+
+  const renderVirtualizedRow = ({
+    data,
+    index,
+    style,
+  }: ListChildComponentProps) => {
+    const baseOption = data[index];
+    /**
+     * react-window calculates the absolute positions of the list items, via an inline style, so
+     * we need to add it to each list item that is being rendered in the viewable list window.
+     * See here if you need to know more: https://github.com/bvaughn/react-window?tab=readme-ov-file#why-is-my-list-blank-when-i-scroll
+     */
+    const optionItem = { ...baseOption, props: { ...baseOption.props, style } };
+    return optionItem;
+  };
+
+  const OuterListboxContext = createContext({});
+
+  const OuterListboxElementType = React.forwardRef<HTMLDivElement>(
+    (props, ref) => {
+      const outerProps = React.useContext(OuterListboxContext);
+      return <div ref={ref} {...props} {...outerProps} />;
+    }
+  );
+
+  function useResetCache(length: number) {
+    const ref = React.useRef<VariableSizeList>(null);
+    useEffect(() => {
+      if (ref.current != null) {
+        ref.current.resetAfterIndex(0, true);
+      }
+    }, [length]);
+    return ref;
+  }
+
+  const ListboxComponent = forwardRef<
+    HTMLDivElement,
+    React.HTMLAttributes<HTMLElement>
+  >(function (props, ref) {
+    const { children, ...other } = props;
+    const itemData: React.ReactElement[] = [];
+    (children as React.ReactElement[]).forEach(
+      (item: React.ReactElement & { children?: React.ReactElement[] }) => {
+        itemData.push(item);
+        itemData.push(...(item.children || []));
+      }
+    );
```

Comment:
Instead of using `.push`, we can we use `.concat` instead? I prefer never using mutations like `.push` because they can easily run us into bugs without anyone noticing.

In this case, we're doing a `.forEach` which I think can easily be replaced by a `.flatMap` with no need to `.push`; just return an array `[item].concat(item.children || [])` or `[item, ...item.children]`.